### ISSUE TITLE
Error: Dynamic server usage: searchParams.userId

### ISSUE THREAD

**Issue by RonAlmog** [vercel/next.js#49182]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Node: 19.3.0
      npm: 9.2.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.3.4
      eslint-config-next: 13.3.4
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://github.com/AntonioErdeljac/next13-airbnb-clone

### To Reproduce

this is a youtube course that I took:
https://www.youtube.com/watch?v=c_-b_isI4vg&t=27105s&ab_channel=CodeWithAntonio
the code is available at his github account:
https://github.com/AntonioErdeljac/next13-airbnb-clone
to reproduce, clone the repo, and build.  it builds fine.
update the next version from 13.2.4. to the latest, 13.3.4 and build.   you will get a build error.

### Describe the Bug

the code runs fine on dev, but when I run build (with next ver 13.3.4), it fails with the error:
Error: Dynamic server usage: searchParams.userId

we are receiving several params from the url, all are optional. no params at all is fine.
the relevant code is in page.tsx, 'Home'.  lines 9 to 14.

### Expected Behavior

should build fine with no error.

### Which browser are you using? (if relevant)

chrome

### How are you deploying your application? (if relevant)

vercel. (failing because of this issue)

**Comment by RonAlmog** [comment]
somebody is whispering in my ear that I should add 
export const dynamic = 'force-dynamic'
at the top of main page.tsx.  is that true? where can I learn more about this? what does it mean?

**Comment by Toshimichi0915** [comment]
https://github.com/AntonioErdeljac/next13-airbnb-clone/issues/12

```jsx:layout.jsx|page.jsx
export const dynamic = 'auto';
// 'auto' | 'force-dynamic' | 'error' | 'force-static'
```

**Comment by RonAlmog** [comment]
@Toshimichi0915 , thank you!
in my case only 'force-dynamic' did the trick. I wish I understood what's behind it. but at least it works. thanks!

**Comment by Toshimichi0915** [comment]
I think you forgot to close the issue

**Comment by bencehusi** [comment]
If I do 'force-dynamic' I get this error: `Error: Dynamic server usage: force-dynamic`.
Has anyone experienced it? What does it mean and what should I do about it?

I'm just reading searchParams, nothing fancy is going on.

**Comment by allestaire** [comment]
Thanks that someone is posting this issue,
I am tired of creating tickets that needs most of the requirements.
I dont know if this framework are fully tested or something, because others are having problem just like me, and it is from time to time

- Eslint only shows at build
- Dynamic pages not working when deploying to Vercel, also vercels deployment documentation is not clear
- I dont know if the creator of this framework encountered or at least tried to deploy this framework to vercel and hopeful they made a well explained documentation on how they do it.
- Also got an article stating how to deploy next.js on vercel, by having this command `yarn build && yarn start` but it does not work.

**Comment by antusaha970** [comment]
> somebody is whispering in my ear that I should add export const dynamic = 'force-dynamic' at the top of main page.tsx. is that true? where can I learn more about this? what does it mean?

it also worked for me

**Comment by StructuralRealist** [comment]
Adding `export const dynamic = 'force-dynamic'` doesn't work in my case. Also, according to the [docs](https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering)

> Using the [searchParams](https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional) [Pages](https://nextjs.org/docs/app/api-reference/file-conventions/page) prop will opt the page into dynamic rendering at request time.

Which suggest that you don't need to add the `dynamic` constant.

In my case it only happens in development mode and only when you navigate to the page or make a change in the code. If you do a full refresh it doesn't show the error.

**Comment by poberbeck** [comment]
I'm experiencing the same issue as @StructuralRealist. My layout is checking the cookies because of the users auth state. So `export const dynamic = 'force-dynamic'` is necessary. However, adding it throws the error `Error: Dynamic server usage: force-dynamic` when navigating to a page with this layout or a hot reload is performed. But only in dev mode.

**Comment by amanverma-11** [comment]
Adding the line =>     export const dynamic = 'force-dynamic'  at the top of page.tsx worked for me as well. Thank you!

**Comment by java-jaydev** [comment]
Adding export const dynamic = 'force-dynamic' worked in my case. Thank you so much!

**Comment by umeshkukreti27** [comment]
got this error 
Unhandled Runtime Error
Error: Dynamic server usage: force-dynamic

**Comment by visualect** [comment]
/* export const dynamic = "force-dynamic" */ worked for me

**Comment by bestickley** [comment]
Next.js docs [here](https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-functions) state:
> Using the [searchParams](https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional) [Pages](https://nextjs.org/docs/app/api-reference/file-conventions/page) prop will opt the page into dynamic rendering at request time.
Isn't this a bug with Next.js? It should automatically pick up on usage of `props.searchParams`, right? We shouldn't have to add the route segment configuration.

**Comment by kyleatcolenso** [comment]
This bug is a still present.

Adding `export const dynamic = 'force-dynamic'` simply rewords the error slightly. A hard refresh does it fix it though. I'm hoping as others have said that it only happens in development mode, so just a slight annoyance for now.

**Comment by poberbeck** [comment]
> I'm experiencing the same issue as @StructuralRealist. My layout is checking the cookies because of the users auth state. So `export const dynamic = 'force-dynamic'` is necessary. However, adding it throws the error `Error: Dynamic server usage: force-dynamic` when navigating to a page with this layout or a hot reload is performed. But only in dev mode.

In my case, I found the problem on my end. 😬
In the root layout, I used (and forgot) a `getStaticProps` for creating language-specific routs. After removing `getStaticProps` the error disappears.

**Comment by StructuralRealist** [comment]
It seems for me the problem is fixed in `13.4.13`
